controller/node_controller.go

```diff
@@ -972,9 +985,52 @@ func (nc *NodeController) syncNodeStatus(pod *corev1.Pod, node *longhorn.Node) e
 		}
 	}
 
+	// invoke environment check (kernel version first, but others may follow
+	kernelRelease, err := lhns.GetKernelRelease()
+	if err != nil {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeEnvironmentCheck, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonKernelVersionCheck),
+			fmt.Sprintf("The kernel version could not be read from pod %s, node %s", pod.Name, pod.Spec.NodeName))
+	} else {
+		isDefective := nc.isDefectiveKernel(kernelRelease)
+		if isDefective {
+			node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeEnvironmentCheck, longhorn.ConditionStatusFalse,
+				string(longhorn.NodeConditionReasonKernelVersionCheck),
+				fmt.Sprintf("Unsupported kernel version %s found on pod %s, node %s", kernelRelease, pod.Name, pod.Spec.NodeName))
+		} else {
+			node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeEnvironmentCheck, longhorn.ConditionStatusTrue, "",
+				fmt.Sprintf("Node %s running kernel %s", pod.Spec.NodeName, kernelRelease))
+		}
+	}
+
 	return nil
 }
 
+// isDefectiveKernel duplicates the logic of longhorn/scripts/environment_check.sh because
+// * It should get the same answer, and
+// * "sort -V" is more accurate than "semver", because Linux kernels are not semver.
+func (nc *NodeController) isDefectiveKernel(kernelRelease string) bool {
```

Comment:
Can we have a unit test of this? Something table driven, preferably that includes good and bad kernels from major distros (e.g. SLES, RHEL, Ubuntu).

I phrase it like "can we" because I think the decision to use `exec.Command` makes it difficult/impossible.

In general, I don't like calling the shell to do a sort, but I do get the logic for wanting to ensure we get the same answer here and in the environment check script.